### ISSUE TITLE
Different logic for punctuation in Ukrainian and English

### ISSUE THREAD

**Issue by WallK** [sspanak/tt9#634]
Hello!
I have a habit of entering the question mark by pressing 1 and left after a word
It still works for English, but the punctuation input layout changed for Ukrainian and it throws me off a bit (:
It uses the new "+" type of adding letters to words (see below)
English after pressing 1:
![en](https://github.com/user-attachments/assets/50645eda-74c2-4a6b-9f5f-2e55ac11c26f)
Ukrainian after pressing 1:
![ua](https://github.com/user-attachments/assets/0570625c-1b08-46f4-a376-f6995c5a55f7)

Is this intentional?
Cheers!

**Comment by sspanak** [comment]
This complication is because of the words with apostrophes. And it seems that I didn't test well enough. You can see the intended behavior, if you type "don't" in English. When you press "1" for an apostrophe, typing another letter ends the previous word and starts a new one. The only possible way to continue typing the same word is to complete "don't" by pressing 8.

To fix the problem, I will: 
1. Make the base word unselectable, to make it possible to select the question mark in one click.
2. End the current word after pressing 1, unless you have selected the apostrophe, in which case it will be possible to append more letters and type some new word. This will be possible only in Ukrainian and Hebrew because of the unique way the apostrophe is used. In all other languages, it will end the word too.
3. Test better 😀
4. Leave the "+" way of representing the combinations. It is more clear than `привіт, | привіт. | привіт' ... | привіт! | привіт?` in my opinion. 

How does this sound?

**Comment by WallK** [comment]
I don't see a very big difference between apostrophe use in Ukrainian and English
The only difference is the after apostrophe letters in English is a short list, and in Ukrainian they are arbitrary
Maybe that's a huge difference for prediction engine (:

I like the plus much more, the repeated word suggestion were a nightmare (:
But at least they let me click left once, like in other languages!

How will the point 2 work with Ukrainian and Hebrew? Will I be able to type "Мар'яна" with 4261952 or will I need to manually select the apostrophe? Because I'm much more happier to press some more left to get question mark than scrolling through to apostrophe

**Comment by sspanak** [comment]
@WallK, have you managed to try the new version? Does it solve the punctuation problems?

**Comment by sspanak** [comment]
Ah, yes, I wanted to ask you about this case.

By default, it always tries to produce 20 suggestions. But the punctuation combinations are less than that, so it adds longer words. I was thinking that, since this is a somewhat special case, when the current digit sequence ends with 1, maybe there should be no longer words?

For example, if you type "426", then you would get: "пас", "мас", "нас" and whatnot. Say, "нас" is the first one. Then, if you press "1", you will get only the exact "4261" combinations: "нас" + `,.'" ... !?` and nothing else. And only after pressing "9" for "я", and if no word filter is applied (by manually selecting a letter before that, or by pressing Filter), you will get "Мар'яна" or similar words.

And this logic will be applied to all languages. I think it is the better choice, even though it will not be possible to type "don't" using "3661" + left, but only using "36614". Actually, now that I explained it to you, I realized "don't" will again require 5 clicks, so typing will not be slower.

**Comment by WallK** [comment]
I think I've never looked for the suggestion before I've pressed all the required buttons, but I could be an exception
Meaning stopping word suggestions after one is perfect for me! The only thing that will bother me here -- inconsistency between this + and ended words with punctuation (but what can you do with a unique language like that, huh?)

"Don't" by 36618 is the only way I knew, the suggestions of words after 1 was a think that irritated me, so I've never tried to use it

I think this is a good default, but maybe a config option is a better idea?
Something along the lines "Suggest longer words after punctuation?" toggle (but with a better wording) (:

**Comment by sspanak** [comment]
Here is [v38.29](https://github.com/sspanak/tt9/releases/download/v1.0-inital/tt9-v38.29.apk) with the latest apostrophe upgrade. Please, test again and let me know if there are any other issues. Obviously, I can't think of all the scenarios when I don't use Ukrainian at all.

> I think this is a good default, but maybe a config option is a better idea?
Something along the lines "Suggest longer words after punctuation?" toggle (but with a better wording) (:

No configuration until someone complains. I don't want to touch that code anymore. :smile:

**Comment by WallK** [comment]
Hey, it works! There are some weird stuff I always forget how to reproduce so I postponed writing a comment
I'll create a new issue instead
Many thanks!
